### PR TITLE
DEVPROD-803 Support uploading files to Parsley via clipboard [minor]

### PR DESCRIPTION
**Pull request by khelif96**
DEVPROD-803

### Description
Adds support to copy and paste files into Parsley from the upload page.
Went against the approach of including a text box because the UX didn't work and pasting a large file into a text box causes the browser to freeze up.

### Screenshots
image

### CODE REVIEW

File: apps/parsley/src/pages/LogDrop/state.ts [hunk 1]
```diff
@@ -9,15 +9,20 @@ type CurrentState =
 type State = {
   currentState: CurrentState;
   file: File | null;
+  text: string | null;
+  type: "file" | "text" | null;
```
Comment (sophstad) [inline apps/parsley/src/pages/LogDrop/state.ts:13]: Could you use an enum here to avoid magic strings?

**Review by sophstad — COMMENTED** [review]
Would it be significant scope creep to accept a pasted file? When I did so it just parsed the filename instead of its contents. No worries if not, this is still a nice improvement

**Comment by khelif96** [comment]
> Would it be significant scope creep to accept a pasted file? When I did so it just parsed the filename instead of its contents. No worries if not, this is still a nice improvement

I think it may be. From what I understand about the feature request, the typical use case is that users run some tests locally, copy the results from their terminal, and paste them into Parsley. Accepting a pasted file just seems like adding a third way to perform the same functionality we already support with drag-and-drop and the file picker. 
As an aside I can't really figure out how to detect the file clip in the first place it keeps appearing as text clipboard contents.

**Review by sophstad — APPROVED** [review]
Nice! Would this perhaps qualify for a minor release given the guidelines we discussed yesterday? If so, I'd remove the Spruce GQL file from this diffset so that Spruce doesn't also get a minor version bump.

**Pull request merged**